test/unit-test/mqtt_agent_utest.c

```diff
@@ -74,21 +110,449 @@ int suiteTearDown( int numFailures )
+    TEST_ASSERT_EQUAL( MQTTBadParameter, mqttStatus );
+
+    /* Success case. */
+    subscribeArgs.pSubscribeInfo = &subscribeInfo;
+    subscribeArgs.numSubscriptions = 1U;
+    mqttStatus = MQTTAgent_Subscribe( &agentContext, &subscribeArgs, &commandInfo );
+    TEST_ASSERT_EQUAL( MQTTSuccess, mqttStatus );
+    TEST_ASSERT_EQUAL_PTR( &command, globalMessageContext.pSentCommand );
+    TEST_ASSERT_EQUAL( SUBSCRIBE, command.commandType );
+    TEST_ASSERT_EQUAL_PTR( &subscribeArgs, command.pArgs );
+    TEST_ASSERT_NULL( command.pCommandCompleteCallback );
+}
+
+/**
+ * @brief Test that MQTTAgent_Unsubscribe() works as intended.
+ */
+void test_MQTTAgent_Unsubscribe( void )
+{
+    MQTTAgentContext_t agentContext;
+    MQTTStatus_t mqttStatus;
+    CommandInfo_t commandInfo = { 0 };
+    Command_t command = { 0 };
+    MQTTSubscribeInfo_t subscribeInfo = { 0 };
+    MQTTAgentSubscribeArgs_t subscribeArgs = { 0 };
+
+    setupAgentContext( &agentContext );
+    pCommandToReturn = &command;
+    commandInfo.cmdCompleteCallback = mockCompletionCallback;
+
+    /* We have already tested the generic cases of NULL parameters above,
+     * so here we only need to test one for coverage. */
+    mqttStatus = MQTTAgent_Unsubscribe( NULL, &subscribeArgs, &commandInfo );
+    TEST_ASSERT_EQUAL( MQTTBadParameter, mqttStatus );
+
+    mqttStatus = MQTTAgent_Unsubscribe( &agentContext, NULL, &commandInfo );
+    TEST_ASSERT_EQUAL( MQTTBadParameter, mqttStatus );
+
+    /* Incorrect subscribe args. */
+    mqttStatus = MQTTAgent_Unsubscribe( &agentContext, &subscribeArgs, &commandInfo );
+    TEST_ASSERT_EQUAL( MQTTBadParameter, mqttStatus );
+
+    subscribeArgs.pSubscribeInfo = &subscribeInfo;
+    subscribeArgs.numSubscriptions = 0U;
+    mqttStatus = MQTTAgent_Unsubscribe( &agentContext, &subscribeArgs, &commandInfo );
+    TEST_ASSERT_EQUAL( MQTTBadParameter, mqttStatus );
+
+    /* Success case. */
+    subscribeArgs.pSubscribeInfo = &subscribeInfo;
+    subscribeArgs.numSubscriptions = 1U;
+    mqttStatus = MQTTAgent_Unsubscribe( &agentContext, &subscribeArgs, &commandInfo );
+    TEST_ASSERT_EQUAL( MQTTSuccess, mqttStatus );
+    TEST_ASSERT_EQUAL_PTR( &command, globalMessageContext.pSentCommand );
+    TEST_ASSERT_EQUAL( UNSUBSCRIBE, command.commandType );
+    TEST_ASSERT_EQUAL_PTR( &subscribeArgs, command.pArgs );
+    TEST_ASSERT_EQUAL_PTR( mockCompletionCallback, command.pCommandCompleteCallback );
+}
+
+/* ========================================================================== */
+
+/**
+ * @brief Test that MQTTAgent_ProcessLoop() works as intended.
+ */
+void test_MQTTAgent_ProcessLoop( void )
+{
+    MQTTAgentContext_t agentContext;
+    MQTTStatus_t mqttStatus;
+    CommandInfo_t commandInfo = { 0 };
+    Command_t command = { 0 };
+
+    setupAgentContext( &agentContext );
+    pCommandToReturn = &command;
+
+    /* We have already tested generic cases of NULL parameters above,
```

Comment:
please be specific about the test in which we are doing this verification as that will be helpful for a future reader.